tests/common-unit/hierarchical-menu/optimistic-ui.ts

```diff
@@ -0,0 +1,118 @@
+import { wait } from '@instantsearch/testutils';
+import {
+  createSearchClient,
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '@instantsearch/mocks';
+import type { HierarchicalMenuSetup } from '.';
+import { fakeAct } from '../common';
+import userEvent from '@testing-library/user-event';
+
+export function createOptimisticUiTests(setup: HierarchicalMenuSetup) {
+  // @TODO: after helper is updated with https://github.com/algolia/algoliasearch-helper-js/pull/925, enable this test
+  describe.skip('optimistic UI', () => {
+    test('is checked immediately with a slow network', async () => {
+      const delay = 100;
+      const margin = 10;
+      const attributes = ['brand'];
+      const options = {
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient: createSearchClient({
+            search: jest.fn(async (requests) => {
+              await wait(delay);
+              return createMultiSearchResponse(
+                ...requests.map(() =>
+                  createSingleSearchResponse({
+                    facets: {
+                      [attributes[0]]: {
+                        Samsung: 100,
+                        Apple: 1000,
+                      },
+                    },
+                  })
+                )
+              );
+            }),
+          }),
+        },
+        attributes,
+      };
+      const env = await setup(options);
+      const { act = fakeAct } = env;
+
+      // wait for initial results
+      await act(async () => {
+        await wait(margin + delay);
+        await wait(0);
+      });
+
+      // before interaction
+      {
+        expect(
+          env.container.querySelectorAll('.ais-HierarchicalMenu-item')
+        ).toHaveLength(2);
+        expect(
+          env.container.querySelectorAll('.ais-HierarchicalMenu-item--selected')
+        ).toHaveLength(0);
+      }
+
+      // select item
+      {
+        const firstItem = env.container.querySelector(
+          '.ais-HierarchicalMenu-link'
+        )!;
```

Comment:
Could we use Testing Library's queries so we can select by label? This way we could also easily assert whether the right refinement is selected.